### DISCUSSION TITLE
Issue setting up incremental sync on custom connector

### DISCUSSION THREAD

**Discussion by SwanSodja** [airbytehq/airbyte#36877] [Connector Questions]
Hi folks,

TL;DR: is it possible to recast a string version of a timestamp as the string version of the corresponding date using jinja expressions allowed in order to properly configure my query parameter?

Here's the details. Any help would be much appreciated.  
I'm attempting to set up a custom connector using the polygon api, and I've got it working for the most part. The part I'm struggling with is, having set up incremental sync to work with the cursor key (this is in epoch time as %ms), how to inject it into the appropriate query parameter as %Y-%m-%d.

here's an example of the format the api requires for url: 
`https://api.polygon.io/v2/aggs/ticker/AAPL/range/1/day/2023-01-09/2023-01-09?adjusted=true&sort=desc&limit=120&apiKey={{ config['api_key'] }}`

here's an excerpt of my yaml where I think I'm doing something wrong:
```
partition_router:
        - type: ListPartitionRouter
          values:
            ...
          cursor_field: stream_ticker
        - type: ListPartitionRouter
          values: '{{ config[''start_date''] }}'  #wrong format from start_date. can we get this from what has been injected into the request body?
          cursor_field: from_date
          request_option:
            inject_into: body_json
            type: RequestOption
            field_name: parameter_output
    transformations:
      - type: AddFields
        fields:
          - path:
              - ticker
            value: '{{ stream_partition.stream_ticker }}'
    incremental_sync:
      type: DatetimeBasedCursor
      cursor_field: t
      cursor_datetime_formats:
        - '%ms'
      datetime_format: '%Y-%m-%d'                 #this is where it gets the correct format
      start_datetime:
        type: MinMaxDatetime
        datetime: '{{ config[''start_date''] }}'
        datetime_format: '%Y-%m-%dT%H:%M:%SZ'
      start_time_option:
        inject_into: body_json                    #this is where I tell it to inject into body just to see what the value is
        type: RequestOption
        field_name: incremental_sync_output
      end_datetime:
        type: MinMaxDatetime
        datetime: '{{ now_utc().strftime(''%Y-%m-%dT%H:%M:%SZ'') }}'
        datetime_format: '%Y-%m-%dT%H:%M:%SZ'
```

and here's where I the result of injecting both the stream_partition value and the incremental sync into the json payload (I did this for troubleshooting:
```
{
  "url": "https://api.polygon.io/v2/aggs/ticker/MSFT/range/1/day/2/2024-03-30",  #the problem is that where it says /2/ in the url, it should be the value of the user-provided variable transformed to YYY-MM-DD format, start_date. this value is 2024-03-28T00:00:00Z
  "parameters": {
    "adjusted": [
      "true"
    ],
    "sort": [
      "desc"
    ],
    "limit": [
      "5000"
    ],
    "apiKey": [
      "****"
    ]
  },
  "body": {
    "parameter_output": "2",                         #I surfaced the returned version of start_date here for trouble shooting 
    "incremental_sync_output": "2024-03-28"          #this is the desired output, surfaced from eslewhere in the code. see below
  },
  "headers": {
    "User-Agent": "python-requests/2.31.0",
    "Accept-Encoding": "gzip, deflate",
    "Accept": "*/*",
    "Connection": "keep-alive",
    "apiKey": "****",
    "Content-Length": "66",
    "Content-Type": "application/json"
  },
  "http_method": "GET"
}
```

[a comment by an automated account is not shown]
